### PR TITLE
Add GitHub Advanced Security integration

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?
Package for ingesting Github Advanced Security (GHAS) alerts provided by [Code Scanning](https://docs.github.com/en/rest/code-scanning) and [Secret Scanning ](https://docs.github.com/en/enterprise-cloud@latest/rest/secret-scanning)
 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

### CODE REVIEW

File: packages/github/_dev/build/docs/README.md [hunk 1]
```diff
@@ -14,4 +12,18 @@ To use this integration, you must be an organization owner, and you must use an
...
+The Code Scanning lets you retrieve all security vulnerabilities and coding errors from a repository setup using Github Advanced Security Code Scanning feature. See [About code scanning](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning) for more details.
+
+To use this integration, you must use an access token with the `security_events` scope for private repos or `public_repo` scope for public repos.
+
+
+## Secret Scanning
+
+The Github Secret Scanning lets you retrieve secret scanning for advanced security alerts from a repository setup using Github Advanced Security Secret Scanning feature. See [About Secret scanning](https://docs.github.com/en/enterprise-cloud@latest/code-security/secret-scanning/about-secret-scanning) for more details.
+
+To use this integration, you must be an administrator for the repository or for the organization that owns the repository, and you must use a personal access token with the `repo` scope or `security_events` scope. For public repositories, you may instead use the public_repo scope.
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/_dev/build/docs/README.md:35]: Added

File: packages/github/data_stream/code_scanning/_dev/test/pipeline/test-common-config.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+fields:
+  tags:
+    - preserve_original_event
+dynamic_fields:
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/code_scanning/_dev/test/pipeline/test-common-config.yml]: Removed

File: packages/github/data_stream/code_scanning/_dev/test/pipeline/test-ghas-code-scan-json.log-expected.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                        "commit_sha": "39406e42cb832f683daa691dd652a8dc36ee8930",
+                        "environment": "{}",
+                        "location": {
+                            "end_column": 18,
+                            "end_line": 917,
+                            "path": "spec-main/api-session-spec.ts",
+                            "start_column": 7,
+                            "start_line": 917
+                        },
+                        "message": {
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/code_scanning/_dev/test/pipeline/test-ghas-code-scan-json.log-expected.json]: Moved it to root level `message`  field as it indicates alert's detailed description

File: packages/github/data_stream/code_scanning/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    timezone: UTC
+    target_field: "github.code_scanning.fixed_at"
+    if: ctx.github.code_scanning?.fixed_at != null
+- date:
+    field: github.code_scanning.dismissed_at
+    formats:
+    - ISO8601
+    timezone: UTC
+    target_field: "github.code_scanning.dismissed_at"
+    if: ctx.github.code_scanning?.dismissed_at != null
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/code_scanning/elasticsearch/ingest_pipeline/default.yml]: Removed the block and set `@timestamp` inside the `date` processor

File: packages/github/data_stream/code_scanning/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+- set:
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/code_scanning/elasticsearch/ingest_pipeline/default.yml]: Removed `set` and `remove` processors in favour of `rename`

File: packages/github/data_stream/code_scanning/elasticsearch/ingest_pipeline/default.yml [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+    ignore_missing: true
+- rename:
+    field: _temp.owner.type
+    target_field: github.code_scanning.repository.owner.type
+    ignore_missing: true
+- rename:
+    field: _temp.owner.site_admin
+    target_field: github.code_scanning.repository.owner.site_admin
+    ignore_missing: true
+- set:
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/code_scanning/elasticsearch/ingest_pipeline/default.yml]: Removed `set` and `remove processors` in favour of rename

File: packages/github/data_stream/code_scanning/fields/fields.yml [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+        The time that the alert was no longer detected and was considered fixed in ISO 8601 format: `YYYY-MM-DDTHH:MM:SSZ`
+
+    - name: dismissed_by
+      type: group
+      description: >
+        Information of user who dismissed the alert
+
+      fields:
+        - name: name
+          type: text
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/code_scanning/fields/fields.yml]: Changed name and other non-descriptive fields to `keyword`

File: packages/github/data_stream/secret_scanning/_dev/test/pipeline/test-common-config.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+fields:
+  tags:
+    - preserve_original_event
+dynamic_fields:
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/secret_scanning/_dev/test/pipeline/test-common-config.yml]: Removed

File: packages/github/data_stream/secret_scanning/elasticsearch/ingest_pipeline/default.yml [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+    timezone: UTC
+    target_field: "github.secret_scanning.resolved_at"
+    if: ctx.github.secret_scanning?.resolved_at != null
+- date:
+    field: github.secret_scanning.push_protection_bypassed_at
+    formats:
+    - ISO8601
+    timezone: UTC
+    target_field: "github.secret_scanning.push_protection_bypassed_at"
+    if: ctx.github.secret_scanning?.push_protection_bypassed_at != null
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/github/data_stream/secret_scanning/elasticsearch/ingest_pipeline/default.yml]: Removed the block and set `@timestamp` inside the `date` processor

**Review by P1llus — APPROVED** [review]

**Pull request merged**